src/Encoder/RailFenceCipher.js

```diff
@@ -0,0 +1,173 @@
+
+import Encoder from '../Encoder'
+
+const meta = {
+  name: 'rail-fence-cipher',
+  title: 'RailFence cipher',
+  category: 'Ciphers',
+  type: 'encoder'
+}
+
+const defaultKey = 2
+
+/**
+ * Encoder brick for RailFence Cipher encoding and decoding
+ */
+export default class RailFenceCipherEncoder extends Encoder {
+  /**
+     * Returns brick meta.
+     * @return {object}
+     */
+  static getMeta () {
+    return meta
+  }
+
+  /**
+     * Constructor
+     */
+  constructor () {
+    super()
+    this.addSettings([
+      {
+        name: 'key',
+        type: 'number',
+        label: 'Key',
+        priority: 10,
+        value: defaultKey,
+        integer: true
+      }
+    ])
+  }
+
+  /**
+     * Performs encode or decode on given content.
+     * @param {Chain} content
+     * @param {boolean} isEncode True for encoding, false for decoding
+     * @return {number[]|string|Uint8Array|Chain} Resulting content
+     */
+  performTranslate (content, isEncode) {
+    const { key } =
+            this.getSettingValues()
+    const n = content.getLength()
+    const result = new Array(n)
+
+    // create the matrix to cipher plain text
+    // key = no. of rows in matrix
+    const rail = new Array(key)
+    for (let i = 0; i < key; i++) {
+      rail[i] = new Array(n)
+    }
+
+    let dirDown = false
+    let row = 0
+    let col = 0
+
+    if (isEncode) {
+      // filling the rail matrix to distinguish
+      // filled spaces from blank ones
+      for (let i = 0; i < n; i++) {
+        // check the direction of flow
+        // reverse the direction if we've just
+        // filled the top or bottom rail
+        if (row === 0 || row === key - 1) {
+          dirDown = !dirDown
+        }
+
+        // fill the corresponding alphabet
+        rail[row][col] = content.getCodePointAt(i)
+
+        col++
+
+        // find the next row using direction flag
+        if (dirDown) {
+          row++
+        } else {
+          row--
+        }
+      }
+      // construct the cipher using the rail matrix
+      let index = 0
+      for (let i = 0; i < key; i++) {
+        for (let j = 0; j < n; j++) {
+          if (rail[i][j]) {
```

Comment:
For documentation purposes always compare explicitly instead of depending on the weird JavaScript comparison table. In this case: `rail[i][j] !== undefined`.